conjure-core/src/main/java/com/palantir/conjure/parser/ConjureParser.java

```diff
@@ -79,6 +91,7 @@ public static ConjureSourceFile parse(File file) {
 
         Map<String, AnnotatedConjureSourceFile> parsed = new HashMap<>();
         Queue<File> toProcess = new ArrayDeque<>(files);
+        Map<File, File> importProvenance = new HashMap<>();
```

Comment:
Instead of a map, let's create a new simple type for the `toProcess` queue which contains both the `File conjureFile()` and `Optional<File> importedBy()`.